tests/helpers/browser/server.js

```diff
@@ -0,0 +1,213 @@
+      var testIncludes = allFiles.filter(function(file) {
+          return /\.js$/i.test(file);
+        }).map(function(file) {
+          return "\"" + file + "\": require(\"" + testDir + "/" + file + "\")";
+        }).join(",\n");
+
+      fs.readFile(__dirname + "/run-all.js.tmpl", function(err, src) {
+        stream.push(
+          String(src).replace(/{{\s*INJECT_TEST_INCLUDES\s*}}/, testIncludes)
+        );
+        stream.push(null);
+      });
+    });
+
+    return stream;
+  }
+};
+
+var build = {
+  "index.html": function(done) {
+    fs.readFile(__dirname + "/index.html", done);
+  },
+
+  "jshint.js": function(done) {
+    buildJSHint("web", function(err, version, src) {
+      done(err, src);
+    });
+  },
+
+  "tests.js": function(done) {
+    var bundle = browserify({
+      insertGlobalVars: {
+        /**
+         * Ensure that the value of `__dirname` uses Unix path separator across
+         * all platforms.
+         *
+         * By default, Browserify defines the `__dirname` global using the
+         * system's native file separator character, but its implementation of
+         * `path.resolve` (as used in `fixture-fs.js`) only includes the Unix
+         * implementation. This inconsistency does not impact file lookup on
+         * either platform, but when path strings are used as key values (as is
+         * the case in `fixture-fs.js`), the separator character must be
+         * consistent.
+         */
+        __dirname: function() {
+          return "'/tests/unit'";
+        }
+      }
+    });
+    var includedFaker = false;
+
+    bundle.require(
+      fs.createReadStream(__dirname + "/fixture-fs.js"),
+      { expose: "fs" }
+    );
+    bundle.add(streams.fixtures());
+    bundle.add(streams.runAllScript(), { basedir: __dirname });
+
+    /**
+     * When Browserify attempts to bundle the JSHint source, inject a simple
+     * "global extraction module"--a CommonJS module that simply exposes the
+     * globally-defined JSHint instance. This ensures that the tests run
+     * against the version of JSHint built with the project's build script (see
+     * above) and not a version dynamically included in the current bundle for
+     * test files.
+     */
+    bundle.transform(function(filename) {
+      var faker;
+
+      if (filename === mainPath) {
+        includedFaker = true;
+        faker = new Stream.Readable();
+        faker._read = faker.write = function() {};
+        faker.push("exports.JSHINT = window.JSHINT;");
+        faker.push(null);
+        return faker;
+      }
+
+      return new Stream.PassThrough();
+    });
+
+    bundle.bundle(function(err, src) {
+      if (err) {
+        done(err);
+        return;
+      }
+
+      if (!includedFaker) {
+        done(new Error(
+          "JSHint extraction module not included in bundled test build."
+        ));
+        return;
+      }
+
+      done(null, src);
+    });
+  }
+};
+
+module.exports = function(port, done) {
```

Comment:
I guess I'm used to node modules generating to a tmp directory, then using an existing http server package to serve that, to avoid writing an http server to run the tests.. but I guess its not that much code... and not having a tmo directory is nice.